docs/contributor_tips.md

```diff
@@ -0,0 +1,58 @@
+# Maintainer Tips
+
+This is an opinionated guide documenting workflows that some members of the team
+have found useful. It is focused on meta-tooling, not on IREE code specifically
+(you will find the latter in the
+[Developer Overview](../developer_overview.md)) It is certainly possible to use
+workflows other than these, but some common tasks, especially for maintainers
+will likely be made easier if you use these flows. It assumes a basic knowledge
+of `git` and GitHub and suggests some specific ways of using it.
+
+## Git Structure
+
+Work primarily from a clone of the repository on your development machine. Any
+local branches named the same as persistent branches from the
+[main repository](https://github.com/google/iree) (currently `main`, `google`,
+and `stable`) are pristine (though potentially stale) copies. You only
+fastforward these to match upstream and otherwise do development on other
+branches. When sending PRs, you push to a different branch on your fork and
+create the PR from there.
+
+### Setup
+
+1. Create a fork of the main repository.
+2. Clone from this repository or the main repository.
+3. Rename your remotes to `upstream` and `fork`, so that it's abundantly clear
+   which is which. This is especially important for maintainers who have write
+   access (so can push directly to the main repository) and admins who have
+   elevated privileges (so can push directly to protected branches). These names
+   are just suggestions, but you might find some scripts that assume the remotes
+   are named like this.
+
+   TODO(gcmn): These are just the names I use. We could poll the team to
+   establish a convention here for the scripts.
+
+4. Create git aliases to easily synchronize your persistent branches with
+   `upstream`.
+
+    ```
+    [alias]
+        update = "!f() { \
+          git checkout ${1?} && git pull upstream ${1?} --ff-only && git submodule update --init && git status; \
+        }; f"
+        sync = ! ([[ -z "$(git status --porcelain)" ]] && git update google && git update main) || (echo "Sync failed" && git status && false)
+    ```
```

Comment:
I'm still not a fan of adding git aliases wrapping common commands and using those aliases throughout scripts. It adds too many steps for people to understand for each project they contribute to.

Reply:
I agreed :-D I think an alias like this is useful on its own though, so is reasonable to suggest (although I'm happy to have a different implementation of it. This one is stupid complicated because of submodules). I think you're right we shouldn't use it in scripts though, but the current scripts are just the ones that used to be in my `~/scripts` directory and seemed better than nothing. Syncing the repo properly is a PITA, so it's useful to have that encapsulated in some command rather than duplicating it in all the scripts. Probably the better way is to write a bash function and source it in the other scripts.

Reply:
These aliases are no longer used in the scripts. Does that make you happier?

Reply:
Yep, though now I'm thinking that most contributors shouldn't need the `git update google` part of that `sync` alias.

Reply:
Updated the suggestion and factored it into a script.